### PR TITLE
Update Datadog config to access temporal data

### PR DESCRIPTION
**Pull request by PopDaph**
Datadog documentation: [Introducing Prometheus support for Datadog Agent 6](https://www.datadoghq.com/blog/monitor-prometheus-metrics/)
Doc from Keith. T. from Temporal: https://keithtenzer.com/temporal/temporal_cloud_observability/

What has been done: 
1/ Generate a new certificate for temporal on GCP. I followed the doc on [Notion](https://www.notion.so/dust-tt/Runbook-Temporal-infra-e065e5ccd53443c1a57118d08c7a19f9) 
2/ Enable observability on Temporal Cloud in: https://cloud.temporal.io/settings/integrations/observability, creating a new prometheus endpoint
3/ Open this PR to edit the datadog config to listen to this new endpoint. 

Non tested, no idea if this will work!

### CODE REVIEW

**Review by fontanierh — DISMISSED** [review]
Thanks for doing this, I have no idea if this is right or not but it _looks_ good. Let's yolo it

**Comment by spolu** [comment]
Looks reasonable :+1: Shouldn't we see the certificate here? Maybe i"m missing it.

**Review by spolu — DISMISSED** [review]

**Comment by PopDaph** [comment]
> Shouldn't we see the certificate here? Maybe i"m missing it.

-> Yes we do and I will also add `temporal_cloud_v0_workflow_cancel_count` to be able to test the integration easier (by cancelling extract workflows)

File: k8s/datadog-values.yml [hunk 1]
```diff
@@ -36,6 +36,8 @@ confd:
         metrics:
           - temporal_cloud_v0_workflow_failed_count
           - temporal_cloud_v0_workflow_timeout_count
+          - temporal_cloud_v0_workflow_cancel_count
+        ssl_ca_cert: /etc/certs/temporal-datadog-cert
```
Comment (PopDaph) [inline k8s/datadog-values.yml:40]: https://github.com/DataDog/integrations-core/blob/master/prometheus/datadog_checks/prometheus/data/conf.yaml.example#L104

**Comment by PopDaph** [comment]
Closing, will do with a service. I redeployed `dust-datadog-agent` from main to reset those changes.

**Pull request closed without merging**